lib/acl.c

```diff
@@ -136,11 +136,13 @@ process_acl(struct gatekeeper_if *iface, unsigned int lcore_id,
 void
 destroy_acls(struct acl_state *astate)
 {
-	unsigned int numa_nodes = get_net_conf()->numa_nodes;
+	struct net_config *net_conf = get_net_conf();
 	unsigned int i;
-	for (i = 0; i < numa_nodes; i++) {
-		rte_acl_free(astate->acls[i]);
-		astate->acls[i] = NULL;
+	for (i = 0; i < net_conf->numa_nodes; i++) {
+		if (net_conf->numa_used[i]) {
```

Comment:
It would be simpler to just test `astate->acls[i] != NULL`. Analogous suggestion applies to other ACL functions such as `register_ipv4_acl()`, `build_ipv4_acls()`, etc. but `init_ipv4_acls()` and `init_ipv6_acls()`.